main/src/ca/uwaterloo/flix/util/Validation.scala

```diff
@@ -295,6 +409,41 @@ object Validation {
                                 (f: (T1, T2, T3) => Validation[U, E]): Validation[U, E] =
     (t1, t2, t3) match {
       case (Success(v1), Success(v2), Success(v3)) => f(v1, v2, v3)
+      case (Success(v1), Success(v2), SuccessWithFailures(v3, e3)) => f(v1, v2, v3) match {
```

Comment:
What in blazes

Reply:
IMO unless there's a significant performance cost, we should really avoid doing this by hand.

Reply:
I proposes to special case for n=1..3 to avoid the overhead of curry and apply etc.

Reply:
Yeah, let's do a clean impl and see what the graph has to say about it.

Reply:
Up to you, I can just revert it :)

Reply:
OK, lets revert it, and see the impact.